### PR TITLE
Make the `user_id` key a configurable property.

### PR DESCRIPTION
**Pull request by TheLazzziest**
## What do these changes do?

Add support of a custom user id key for session storage.

## Related issue number

#468 

## TODO
[x] Revert requirement files
[x] Revert tox
[x] Revert travis

### CODE REVIEW

**Comment by maxcountryman** [comment]
This seems like it introduces a lot of extraneous structural changes to the codebase, in particular with testing and requirements, that I'd like to separate into a different PR.

**Comment by TheLazzziest** [comment]
As for requirement files, I agree. It ought to be placed in a separate pr. But, separating tests into a different one doesn't look like a good decision. I would prefer to leave it here, because it's directly related to the current feature.

The structure of tests doesn't affect on usage of the library, but instead, improve code quality by adding an additional context to tests.

**Comment by maxcountryman** [comment]
I agree that tests should be part of this change, but I’m concerned about structural changes. If we can clean up the other changes then I think it’ll be a little easier to review and make sure we’re not introducing dramatic changes for the sake of a relatively small feature.

**Comment by TheLazzziest** [comment]
So, you want me to revert tests to a single module, right?

As for structural changes, I didn't touch the core in terms of the structure, so there shouldn't be any side effects. Moreover, if I messed up with test structure or anything else which had an impact on a code functionality, unit tests should reflect this.

Everything else is related only to logical and code style perspectives.

**Comment by TheLazzziest** [comment]
I leave the TODO list above. Can you validate? Maybe I missed something. I want to clarify what you expect to be done.

**Comment by alanhamlett** [comment]
FYI, I'm working on a PR that makes all session keys configurable.

**Comment by TheLazzziest** [comment]
@alanhamlett , that's great. Why don't you take my changes to your pr? Or you will do it in another way?

**Comment by TheLazzziest** [comment]
@maxcountryman , hi. I did amends. Is this pr still relevant?

File: .gitignore [hunk 1]
```diff
@@ -1,7 +1,9 @@
 !.gitignore
 
+.idea/
```
Comment (alanhamlett) [inline .gitignore]: These aren't Python-specific and should go in your personal global gitignore config file.

Comment (TheLazzziest) [reply .gitignore]: You're right. I remove then entry.

Comment (TheLazzziest) [reply .gitignore]: @alanhamlett , what's next ?

**Pull request closed without merging**